src/Core/src/Handlers/HybridWebView/HybridWebViewHandler.cs

```diff
@@ -339,10 +339,17 @@ await handler.InvokeAsync(nameof(IHybridWebView.EvaluateJavaScriptAsync),
 
 			var stringResult = await callback.Task;
 
-			if (stringResult is null)
+			// if we are not looking for a return object, then return null
+			if (invokeJavaScriptRequest.ReturnTypeJsonTypeInfo is null)
 			{
 				invokeJavaScriptRequest.SetResult(null);
 			}
+			// if there is no result or if the result was null/undefined, then reat it as null
+			else if (stringResult is null || stringResult == "null" || stringResult == "undefined")
+			{
+				invokeJavaScriptRequest.SetResult(null);
+			}
+			// if we are expecting a result, then deserialize what we have
 			else
```

Comment:
This is the real fix, if no type info is provided and/or the return type is null/undefined (from JS) then we just return null in .NET.